treedoc/traversal.py

```diff
@@ -249,6 +251,7 @@ def recurse_to_object(self, obj) -> bool:
 
     def _search(self, *, obj, stack=None, final_node_at_depth=None):
         """
+        Missing docstring.
```

Comment:
@tommyod : Could you provide a short docstring here, and I'll add it?

Also, I'm confused about the use of `*` here. My understanding is that `*` limits the users ability to add positional arguments beyond the point where `*` is specified, as described in [PEP 3102](https://www.python.org/dev/peps/pep-3102/). Should the function declaration be

`def _search(self, obj, *, stack=None, final_node_at_depth=None)`

or am I missing something?

Reply:
No correct answer here, both options work. 

The `_search` method is "private" (only to be called by other methods, not by a user) so I chose to restrict every argument to be positional for robustness. In `search` however I'm okay with the user writing `search(list)` instead of having to write `search(obj=list)`, which is tiresome for the user. That's why they're different.